vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -438,469 +558,2250 @@ func TestAccResourceVSphereVirtualMachine(t *testing.T) {
+// machine's primary NIC has the given IP address and netmask assigned to it,
+// and that the appropriate gateway is present.
+//
+// This uses VMware tools to check this, so it needs to be installed on the
+// guest.
+func testAccResourceVSphereVirtualMachineCheckNet(expectedAddr, expectedPrefix, expectedGW string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		res, err := strconv.Atoi(expectedPrefix)
+		if err != nil {
+			return fmt.Errorf("bad: %s", err)
+		}
+		expectedPrefixInt := int32(res)
+
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		var v4gw, v6gw net.IP
+		for _, s := range props.Guest.IpStack {
+			if s.IpRouteConfig != nil {
+				for _, r := range s.IpRouteConfig.IpRoute {
+					switch r.Network {
+					case "0.0.0.0":
+						v4gw = net.ParseIP(r.Gateway.IpAddress)
+					case "::":
+						v6gw = net.ParseIP(r.Gateway.IpAddress)
+					}
+				}
+			}
+		}
+		for _, n := range props.Guest.Net {
+			if n.IpConfig != nil {
+				for _, addr := range n.IpConfig.IpAddress {
+					ip := net.ParseIP(addr.IpAddress)
+					if !ip.Equal(net.ParseIP(expectedAddr)) {
+						continue
+					}
+					if addr.PrefixLength != expectedPrefixInt {
+						continue
+					}
+					var mask net.IPMask
+					if ip.To4() != nil {
+						mask = net.CIDRMask(int(addr.PrefixLength), 32)
+					} else {
+						mask = net.CIDRMask(int(addr.PrefixLength), 128)
+					}
+					switch {
+					case ip.Mask(mask).Equal(v4gw.Mask(mask)):
+						if net.ParseIP(expectedGW).Equal(v4gw) {
+							return nil
+						}
+					case ip.Mask(mask).Equal(v6gw.Mask(mask)):
+						if net.ParseIP(expectedGW).Equal(v6gw) {
+							return nil
+						}
+					}
+				}
+			}
+		}
+
+		return fmt.Errorf("could not find IP %s/%s, gateway %s", expectedAddr, expectedPrefix, expectedGW)
+	}
+}
+
+// testAccResourceVSphereVirtualMachineCheckStaticMACAddr is a check to look
+// for the MAC address defined in the static MAC address test on the first
+// network interface.
+func testAccResourceVSphereVirtualMachineCheckStaticMACAddr() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		props, err := testGetVirtualMachineProperties(s, "vm")
+		if err != nil {
+			return err
+		}
+		l := object.VirtualDeviceList(props.Config.Hardware.Device)
+		devices := l.Select(func(device types.BaseVirtualDevice) bool {
+			if _, ok := device.(types.BaseVirtualEthernetCard); ok {
+				return true
+			}
+			return false
+		})
+		if devices[0].(types.BaseVirtualEthernetCard).GetVirtualEthernetCard().AddressType != string(types.VirtualEthernetCardMacTypeManual) {
+			return fmt.Errorf("first network interface is not set to manual address type")
+		}
+		actual := devices[0].(types.BaseVirtualEthernetCard).GetVirtualEthernetCard().MacAddress
+		expected := testAccResourceVSphereVirtualMachineStaticMacAddr
+		if expected != actual {
+			return fmt.Errorf("expected MAC address to be %s, got %s", expected, actual)
```

Comment:
_minor_ these probably want to be `%q` rather than `%s`?

Reply:
Updated!